### ISSUE TITLE
Directional gradient with angle

### ISSUE THREAD

**Issue by erf** [wez/wezterm#2023]
**Is your feature request related to a problem? Please describe.**

I'd like to express a gradient that moves diagonally from top left to bottom right linearly across the window, like expressed in [these examples](https://digitalsynopsis.com/design/beautiful-gradient-color-palettes/).

Or more generally could be expressed with a direction angle.

**Describe the solution you'd like**

``` Lua
window_background_gradient = {
		orientation = { Direction = { angle = 135 } }
}
```

This would make a gradient in a direction from start angle and through the window where, where the normal on that direction vector will always have the same color.

**Describe alternatives you've considered**

Radial orientation or orientation "Diagonally", but i think a "Direction" angle (or radian) expresses this better.

**Additional context**

Thanks so much for this gradient ! My terminal is looking pretty sweet now :)

**Comment by wez** [comment]
I'd like to see this. I think it is conceptually simple, if you have a good handle on geometry. I'd love to see a PR!

**Comment by erf** [comment]
Cool! Super busy at the moment, but i can take a look at it when i find time and maybe brush off some linear algebra dust

**Comment by erf** [comment]
Fixed in #2038
